test/testsuppressions.cpp

```diff
@@ -1203,9 +1203,14 @@ class TestSuppressions : public TestFixture {
         ASSERT_EQUALS("", settings.nomsg.addSuppressionLine("uninitvar"));
         settings.exitCode = 1;
 
-        const char code[] = "int f() { int a; return a; }";
-        ASSERT_EQUALS(0, cppCheck.check("test.c", code)); // <- no unsuppressed error is seen
```

Comment:
hmm.. this means that the file must exist even if the code is provided..

in the democlient we don't check actual files neither.
https://cppcheck.sourceforge.io/demo/

it's really an unusual edge case but still..

I have the feeling that if you put the check in: `unsigned int CppCheck::check(const FileSettings &fs)` then this test will still be happy. And the democlient will be happy. And importing projects with missing mismatching files will not crash..

Imho A test case in TestCppcheck would be nice.